src/config/rate-limiting-error.ts

```diff
@@ -0,0 +1,12 @@
+export class RateLimitingError extends Error {
+	/**
+	 * The value of the x-ratelimit-reset header, i.e. the epoch seconds when the rate limit is refreshed.
+	 */
+	rateLimitReset: number;
+
+	constructor(resetEpochSeconds: number) {
+		super("rate limiting error");
+		this.rateLimitReset = resetEpochSeconds;
+		Object.setPrototypeOf(this, RateLimitingError.prototype);
```

Comment:
I am pretty sure this `setPrototypeOf` is pre-class js era stuff. 
We are using `class`, so we probably don't need this one.

------------

I do understand this is from existing code, not a new change in this PR, so I am happy to leave it for another PR.

Reply:
Made it simpler for all and removed it now. the tests that was relying on it had a test for old(this one) and new so was a pretty straight forward remove.

Reply:
even better.